ddlitlab2024/dataset/recording2mcap.py

```diff
@@ -0,0 +1,308 @@
+import json
+import shutil
+import sys
+from pathlib import Path
+
+from sqlalchemy.orm import Session
+
+from ddlitlab2024.dataset import logger
+from ddlitlab2024.dataset.models import Recording, stamp_to_seconds_nanoseconds
+
+try:
+    import rosbag2_py
+    from builtin_interfaces.msg import Time
+    from geometry_msgs.msg import Quaternion
+    from rclpy.serialization import serialize_message
+    from sensor_msgs.msg import Image, JointState
+    from std_msgs.msg import Header, String
+except ImportError:
+    logger.error(
+        "Failed to import ROS 2 packages. These are necessary to convert recordings to .mcap files. "
+        "Make sure ROS 2 installed and sourced."
+    )
+    sys.exit(1)
+
+
+def get_recording(db: Session, recording_id_or_filename: str | int) -> Recording:
+    """Get the recording from the input string or integer
+
+    param db: The database
+    param recording_id_or_filename: The recording ID or original filename
+    raises ValueError: If the recording does not exist
+    return: The recording
+    """
+    if isinstance(recording_id_or_filename, int) or recording_id_or_filename.isdigit():
+        # Verify that the recording exists
+        recording_id = int(recording_id_or_filename)
+        recording = db.query(Recording).get(recording_id)
+        if recording is None:
+            raise ValueError(f"Recording '{recording_id}' not found")
+        return recording
+    elif isinstance(recording_id_or_filename, str):
+        recording = db.query(Recording).filter(Recording.original_file == recording_id_or_filename).first()
+        if recording is None:
+            raise ValueError(f"Recording with original filename '{recording_id_or_filename}' not found")
+        return recording
+    else:
+        raise TypeError("Recording ID must be an integer or string")
+
+
+def get_writer(output: Path) -> rosbag2_py.SequentialWriter:
+    """Get the mcap writer.
+
+    param output: The output mcap file
+    return: The mcap writer
+    """
+    if output.exists():
+        # Ask the user if they want to overwrite the existing file
+        if not input(f"Output directory '{output}' already exists. Overwrite? (y/n): ").lower().startswith("y"):
+            logger.info("Exiting")
+            sys.exit(0)
+        # Remove the existing directory
```

Comment:
Do you think comments like these are helpful? 
I'm personally not a big proponent of comments in general, and think they should only ever be used to explain why something is done and not what is done (especially when it is obvious)

Reply:
Do these comments bother you? Why?

I agree, describing the Why is more important (and often missing) than the What.
But I still see value in these comments, as they describe the intent of the code block. That means, I can quickly understand the flow and actions of the code without reading code which is noisier and requires more experience. Reading one comment spares me from parsing one block in my head.

In this case:
I very rarely use `shutil` and `rmtree` is not the most intuitive name. Of course, I could assume its function considering its input, but when still unsure it requires to read the docs.

Alternatively, one could demand each code-block with such a comment should be its method with proper documentation. This seems silly for a one-liner. For longer blocks, I would normally agree, but this is just a small utility tool, where I don't think it's worth the effort. That's why I use these comments as a middle ground between proper clean functions or nothing.